### PR TITLE
Added 3 images for CI utilities

### PR DESCRIPTION
**Pull request by ianbtr**
The images added support the following CI capabilities:
1. Compiling C++ applications with Boost and CMake
2. Creating TAR.GZ archives
3. Running Python in a LaTeX environment

### CODE REVIEW

File: cxx-extras-Alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM alpine
```
Comment (nicklauslittle-gov) [inline cxx-extras-Alpine/Dockerfile:1]: The CI builds for the target `deploy`, so you will need to add it to all the `Dockerfile`s.

```suggestion
FROM alpine AS deploy
```

Comment (ianbtr) [reply cxx-extras-Alpine/Dockerfile]: Should be fixed now

**Comment by nicklauslittle-gov** [comment]
Please squash.

**Comment by nicklauslittle-gov** [comment]
> Please squash.

I've squashed and pushed this.

**Pull request closed without merging**